server/src/api/customers.ts

```diff
@@ -83,4 +84,38 @@ customerRouter.post(
   }
 );
 
+/**
+ * Handles the PATCH requests to modify an existing customer with the specified customerId.
+ * If successful, returns the modified customer in the response body with code 200.
+ */
+customerRouter.patch(
+  '/:customerId',
+  customerAuth,
+  validateAndFormatPhoneNumber(
+    (body: Partial<CustomerPayload>) => body.contactNumber,
+    (body: Partial<CustomerPayload>, phoneNumber: string) =>
+      (body.contactNumber = phoneNumber)
+  ),
+  async (req: Request, res: Response, next: NextFunction) => {
+    const {customerId: customerIdStr} = req.params;
+    const customerId = Number(customerIdStr);
+    const customerData = req.body;
+
+    try {
+      if (Number.isNaN(customerId)) {
+        throw new Error('Invalid customerId.');
+      }
+
+      const modifiedCustomer = await customerService.updateCustomer(
+        customerId,
+        customerData
+      );
+      res.status(200).send(modifiedCustomer);
```

Comment:
I think there should be 'Content-Location' header here as well, since we are returning an object that can be retrieved using another method.

Reply:
@karenfrilya97 
According to https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html section **14.14 Content-Location** it doesn't seem necessary especially since it is the exact same url that the user has requested for?

Reply:
@hzxcaryn Oh okay! Thanks for clarifying.